iop/cdvd/cdfs/src/main.c

```diff
@@ -0,0 +1,469 @@
+#include <stdio.h>
+#include <ioman.h>
+#include <iox_stat.h>
+#include <sysclib.h>
+
+#include "cdfs_iop.h"
+
+// 16 sectors worth of toc entry
+#define MAX_FILES_PER_FOLDER 256
+#define MAX_FILES_OPENED 16
+#define MAX_FOLDERS_OPENED 16
+#define MAX_BITS_READ 16384
+
+#define UNIT_NAME "cdfs"
+
+struct fdtable
+{
+    iop_file_t *fd;
+    int fileSize;
+    int LBA;
+    int filePos;
+};
+
+struct fodtable
+{
+    iop_file_t *fd;
+    int files;
+    int filesIndex;
+    struct TocEntry entries[MAX_FILES_PER_FOLDER];
+};
+
+// File Descriptors
+static struct fdtable fd_table[MAX_FILES_OPENED];
+static int fd_used[MAX_FILES_OPENED];
+
+// Folder Descriptors
+static struct fodtable fod_table[MAX_FOLDERS_OPENED];
+static int fod_used[MAX_FOLDERS_OPENED];
+
+// global variables
+static int lastsector;
+static int last_bk = 0;
+
+/***********************************************
+*                                              *
+*             DRIVER FUNCTIONS                 *
+*                                              *
+***********************************************/
+
+static int fio_init(iop_device_t *driver)
+{
+    printf("CDFS Filesystem v2\n");
```

Comment:
This is the driver description. It would be nice if it matches the description passed to AddDrv below. Perhaps put the description in a define at the top, just like the #define UNIT_NAME.

Reply:
Done!